src/job/job.service.ts

```diff
@@ -30,4 +31,30 @@ export class JobService {
       return { ...job, jobSequence: jobSequence + 1 };
     });
   }
+
+  async createTransferAssets(dto: CreateTransferAssetsDto) {
+    return await this.prismaService.$transaction(async (tx) => {
+      const isExisting = await tx.job.findUnique({
+        where: { id: dto.id },
+      });
+      if (isExisting) throw new BadRequestException('Job already exists');
+
+      const address = dto.avatarAddress ?? dto.agentAddress;
+      const job = await tx.job.create({
+        data: {
+          id: dto.id,
+          actionType: 'TRANSFER_ASSETS',
+          address,
+          ticker: dto.item.ticker,
+          amount: dto.item.amount,
+        },
+      });
+
+      const jobSequence = await this.prismaService.job.count({
+        where: { createdAt: { lt: job.createdAt }, processedAt: null },
+      });
```

Comment:
Well, is it okay to use `this.prismaService` instead of `tx` in this context?

Reply:
It should be fixed. thank you for alert me!

Reply:
```suggestion
      const jobSequence = await tx.job.count({
        where: { createdAt: { lt: job.createdAt }, processedAt: null },
      });
```